src/stores/examples/index.ts

```diff
@@ -0,0 +1,212 @@
+import { busDispatch } from '@pivanov/event-bus';
+import { toast } from 'react-hot-toast';
+import { create } from 'zustand';
+
+import { snippets } from '@constants/snippets';
+import gistService from '@services/examplesService';
+import { createSelectors } from 'src/stores/createSelectors';
+
+import type { ICodeExample } from '@custom-types/codeSnippet';
+import type {
+  IDeleteExampleModalClose,
+  IEditExampleInfoModalClose,
+  IUploadExampleModalClose,
+} from '@custom-types/eventBus';
+
+interface UploadCustomExampleData {
+  code: string;
+  description: string;
+  exampleName: string;
+}
+
+interface StoreInterface {
+  examples: ICodeExample[];
+  selectedExample: ICodeExample;
+  loading: {
+    isGettingExamples: boolean;
+    isUploading: boolean;
+    isSavingInfo: boolean;
+    isSavingContent: boolean;
+    isDeleting: boolean;
+  };
+  actions: {
+    resetStore: () => void;
+    uploadExample: (data: UploadCustomExampleData) => void;
+    loadExampleContent: (id: string, type: string) => void;
+    updateExampleInfo: (id: string, exampleName: string, description: string) => void;
+    updateExampleContent: (data: string) => void;
+    getExamples: () => void;
+    deleteExample: (id: string) => void;
+  };
+  init: () => void;
+
+}
+
+const initialState: Omit<StoreInterface, 'actions' | 'init'> = {
+  examples: [],
+  selectedExample: {
+    id: '',
+    name: '',
+    description: '',
+    code: '',
+  },
+  loading: {
+    isGettingExamples: false,
+    isUploading: false,
+    isSavingInfo: false,
+    isSavingContent: false,
+    isDeleting: false,
+  },
+};
+
+const baseStore = create<StoreInterface>()((set, get) => ({
+  ...initialState,
+  actions: {
+    uploadExample: async (data) => {
+      set({ loading: { ...get().loading, isUploading: true } });
+
+      if (!data.code || !data.description || !data.exampleName) {
+        console.log(null, 'Invalid input data: Missing code, description, or exampleName');
+        set({ loading: { ...get().loading, isUploading: false } });
+        return;
+      }
+
+      try {
+        const newExamples = await gistService.uploadExample(data);
```

Comment:
Why we should return all the examples once we upload new one. Also shouldn't this be called createExample ?

Reply:
done